app/models/miq_server/worker_management/monitor/kill.rb

```diff
@@ -4,14 +4,37 @@ module MiqServer::WorkerManagement::Monitor::Kill
   def kill_all_workers
     return unless self.is_local?
 
-    killed_workers = []
-    miq_workers.each do |w|
-      if MiqWorker::STATUSES_CURRENT_OR_STARTING.include?(w.status)
-        w.kill
-        worker_delete(w.pid)
-        killed_workers << w
+    remove_unknown_workers
+    remove_workers
+  end
+
+  private
+
+  def remove_unknown_workers
+    # We try, but forget to write migrations to delete orphaned worker class rows, causing this method
+    # to blow up and cause the server to continually fail to start.
+    # This is called very early from evm_application and ensures we attempt to call MiqWorker#kill
+    # to remove the problematic rows, and their in flight queue messages.
+    bad_workers = miq_workers.where.not(:type => MiqWorker.descendants.map(&:name))
+    if bad_workers.size > 0
+      begin
+        MiqWorker.inheritance_column = "__disabled"
+        remove_workers(bad_workers)
+      ensure
+        MiqWorker.inheritance_column = "type"
       end
     end
+  end
+
+  def remove_workers(workers = miq_workers)
+    killed_workers = []
+
+    workers.each do |w|
+      w.kill_process if MiqWorker::STATUSES_CURRENT_OR_STARTING.include?(w.status)
+      w.destroy
+      worker_delete(w.pid)
+      killed_workers << w
```

Comment:
@gtanzillo can you review this change?

Previously, we only ran `MiqWorker#kill` which killed the process AND destroyed the worker row, removed it from the `@workers` list of workers the server was managing, and removed it from the MiqServer list of miq_workers if the worker was started, starting, creating, etc. but not stopped, aborted, etc.

Now, we do this for stopped, aborted, etc. workers too but we do not try to kill the process.